src/main/java/ch/njol/skript/expressions/ExprNoDamageTicks.java

```diff
@@ -18,78 +18,80 @@
  */
 package ch.njol.skript.expressions;
 
-import org.bukkit.entity.LivingEntity;
-import org.bukkit.event.Event;
-import org.eclipse.jdt.annotation.Nullable;
-
+import ch.njol.skript.Skript;
 import ch.njol.skript.classes.Changer.ChangeMode;
-import ch.njol.skript.doc.Description;
-import ch.njol.skript.doc.Examples;
-import ch.njol.skript.doc.Name;
-import ch.njol.skript.doc.Since;
+import ch.njol.skript.doc.NoDoc;
 import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
 import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+import org.jetbrains.annotations.ApiStatus;
 
-@Name("No Damage Ticks")
-@Description("The number of ticks that an entity is invulnerable to damage for.")
-@Examples({"on damage:",
-		"	set victim's invulnerability ticks to 20 #Victim will not take damage for the next second"})
-@Since("2.5")
+@NoDoc
+@Deprecated
+@ApiStatus.ScheduledForRemoval(inVersion = "2.10")
 public class ExprNoDamageTicks extends SimplePropertyExpression<LivingEntity, Long> {
 	
 	static {
 		register(ExprNoDamageTicks.class, Long.class, "(invulnerability|invincibility|no damage) tick[s]", "livingentities");
 	}
-
+	
 	@Override
-	public Long convert(LivingEntity e) {
-		return (long) e.getNoDamageTicks();
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		if (!super.init(exprs, matchedPattern, isDelayed, parseResult))
+			return false;
+		
+		Skript.warning("Switch from deprecated 'no damage ticks' to 'no damage duration', as the deprecated expression will be removed in 2.8.");
```

Comment:
Could be good to use the deprecation warning: https://github.com/SkriptLang/Skript/pull/6549